src/cmap/connection_pool.ts

```diff
@@ -289,9 +336,25 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
    * Pool reset is handled by incrementing the pool's generation count. Any existing connection of a
    * previous generation will eventually be pruned during subsequent checkouts.
    */
-  clear(): void {
-    this[kGeneration] += 1;
-    this.emit('connectionPoolCleared', new ConnectionPoolClearedEvent(this));
+  clear(serviceId?: ObjectId): void {
+    if (this.loadBalanced && serviceId) {
+      const sid = serviceId.toHexString();
+      const generation = this.serviceGenerations.get(sid);
+      // Only need to worry if the generation exists, since it should
+      // always be there but typescript needs the check.
+      if (generation == null) {
+        throw new MongoDriverError('Service generations are required in load balancer mode.');
+      } else {
+        // Increment the generation for the service id.
+        /* eslint no-console: 0 */
```

Comment:
Just noting so we don't forget to remove this

Reply:
Removed.